qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -57,3 +62,71 @@ def post(self):
         else:
             raise HTTPError(400, 'Unknown sample template action: %s' % action)
         self.write(result)
+
+
+def _build_sample_summary(study_id, user_id):
```

Comment:
Remember that HTML tables are built row by row, not column by column, so we would essentially still need a dictionary with sample as key and, at best, the `<td>yes/no</td>` for each prep template. This is effectively what the DataFrame gives us, with optimizations to the column merging for each prep template and HTML generation.